modules/core/shared/src/main/scala-3/io/circe/derivation/Configuration.scala

```diff
@@ -0,0 +1,44 @@
+package io.circe.derivation
+
+object Configuration:
+  val default: Configuration = Configuration()
+  val empty: Configuration = Configuration(useDefaults = false)
+
+/**
+ * Configuration allowing customisation of the JSON produced when encoding, or expected when decoding.
+ *
+ * @param transformMemberNames Transforms the names of any case class members in the JSON allowing, for example,
+ *                             formatting or case changes.
+ * @param useDefaults Whether to allow default values as specified for any case-class members.
+ * @param discriminator Optional key name that, when given, will be used to store the name of the constructor of an ADT
+ *                      in a nested field with this name. If not given, the name is instead stored as a key under which
+ *                      the contents of the ADT are stored as an object.
+ * @param transformConstructorNames Transforms the value of any constructor names in the JSON allowing, for example,
+ *                                  formatting or case changes.
+ * @param strictDecoding Whether to fail when superfluous fields are found.
+ */
+case class Configuration(
+  transformMemberNames: String => String = Predef.identity,
```

Comment:
Would not it make more sense to define transformations as two functions, i.e. separate functions for encoding and decoding?

Reply:
Not sure If I understand what you are getting at. You want a transformMemberNames function that will be used when encoding and another one (also transforming the member names) used when decoding? If those two functions don't agree then doing `decode(encode(obj))` will never work.

Reply:
Hm, I think I misunderstood the way it works.